### PR TITLE
[core] Optimize Query Rendered Features API

### PR DESCRIPTION
**Pull request by pozdnyakov**
This PR filters out invisible layers before quering features
(rather than after, like it happened before).

With this change `API_queryRenderedFeaturesAll` performance test
passes almost two times faster.

Fixes: #15207

### CODE REVIEW

File: src/mbgl/renderer/render_layer.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ bool RenderLayer::hasRenderPass(RenderPass pass) const {
 }
 
 bool RenderLayer::needsRendering() const {
-    return passes != RenderPass::None
```
Comment (astojilj) [inline src/mbgl/renderer/render_layer.cpp]: Now, RenderLayer::passes and evaluatedProperties->renderPasses is the same (as set i evaluate()). Why the change?

Comment (pozdnyakov) [reply src/mbgl/renderer/render_layer.cpp]: leftover, thanks for the catch!
